src/track/beats.h

```diff
@@ -1,49 +1,64 @@
-#ifndef BEATS_H
-#define BEATS_H
+#pragma once
 
+#include <QByteArray>
+#include <QList>
+#include <QMutex>
 #include <QObject>
 #include <QString>
-#include <QList>
-#include <QByteArray>
-#include <QSharedPointer>
+#include <QVector>
+#include <memory>
 
-#include "util/memory.h"
+#include "proto/beats.pb.h"
+#include "track/bpm.h"
+#include "track/frame.h"
 #include "util/types.h"
 
-namespace {
-    double kMaxBpm = 500;
-}
-
 namespace mixxx {
 
 class Beats;
-typedef QSharedPointer<Beats> BeatsPointer;
+class BeatIterator;
+using BeatsPointer = std::shared_ptr<Beats>;
```

Comment:
Please stick to the traditional `typedef` unless we need to introduce aliases for template types.